File: graphJava/Digraph.java
import java.util.LinkedList;

/**
 * Digraph
 */
public class Digraph {

    private final int V;
    public LinkedList<Integer>[] adj;

    public Digraph(int V){

        this.V = V;
        adj = (LinkedList<Integer>[]) new LinkedList[V];

        for(int i = 0 ; i < V ; i++)
            adj[i] = new LinkedList<Integer>();

    }

    public void addEdge(int v , int w){

        adj[v].add(w);

    }

    public int V(){

        return this.V;

    }

    public Iterable<Integer>adj(int v){

        return adj[v];

    }

    public void printGraph(){

        for (int i = 0; i < V; i++) {
            for (Integer w : adj[i]) {
                System.out.println(i + " - " + w);
            }
        }

    }

    public static void main(String[] args) {
        
        System.out.println("Hello World");

        Digraph G = new Digraph(7);

        G.addEdge(0,2);
        G.addEdge(0,5);
        G.addEdge(0, 4);
        G.addEdge(6, 0);
        G.addEdge(3, 6);
        G.addEdge(3, 4);
        G.addEdge(3, 5);
        G.addEdge(3, 4);
        G.addEdge(6, 4);
        G.addEdge(1, 4);

        G.printGraph();

    }

}

File: graph/digraph.cpp
#include<bits/stdc++.h>
#include"digraph.hpp"

using namespace std;

digraph::digraph(int V){

    this->V = V;
    adj.resize(V);

}

void digraph::addEdge(int v , int w){

    adj[v].push_back(w);

}

int digraph::getV(){

    return V;

}

void digraph::printGraph(){

    for(int i = 0 ; i < this->V ; i++){

        cout<<i<<" ";

        for(int j = 0 ; j < adj[i].size() ; j++)
            cout<<adj[i][j]<<" ";

        cout<<endl;

    }

}